src/type/validate.js

```diff
@@ -578,6 +581,64 @@ function validateInputFields(
   });
 }
 
+function validateInputObjectCircularReferences(
+  context: SchemaValidationContext,
+): void {
+  // Modified copy of algorithm from 'src/validation/rules/NoFragmentCycles.js'.
```

Comment:
Since this algorithm is not trivial I will try to extract duplicating code into `jsutils`